CRM/Core/DAO.php

```diff
@@ -1038,16 +1038,17 @@ public static function writeRecord(array $record): CRM_Core_DAO {
     }
     $entityName = CRM_Core_DAO_AllCoreTables::getEntityNameForClass($className);
 
-    // For legacy reasons, empty values would sometimes be passed around as the string 'null'.
-    // The DAO treats 'null' the same as '', and an empty string makes a lot more sense!
-    // For the sake of hooks, normalize these values.
```

Comment:
I don't think this comment should be moved, it still applies to the lines below.